contents/docs/migrate/migrate-between-posthog-instances.mdx

```diff
@@ -42,4 +41,14 @@ Follow the instructions on the [PostHog migrate metadata repo](https://github.co
     ts-node --source [posthog instance you want to migrate from] --sourcekey [personal api key for that instance] --destination [posthog instance you want to migrate to.] --destinationkey [personal api key for destination instance]
     ```
 
-For more information on the options see the [repo's readme](https://github.com/PostHog/posthog-migrate-meta)
\ No newline at end of file
+For more information on the options see the [repo's readme](https://github.com/PostHog/posthog-migrate-meta)
+
+## Migrating your custom apps if you are moving from self-hosted to cloud
+
+Have a custom PostHog app on your self-hosted instance that you'd like to move to the cloud? There are several options:
+1. (Recommended) If you can make your app generalizable enough that others can benefit then [submit your app](https://posthog.com/docs/apps/build/tutorial#submitting-your-app) to the store.
+   - To make it generalizable you'll want to convert anything which is specific to your configuration into a plugin.json config value
+2. If you can't make your app generalizable, you have several options:
+   1. If the app was transforming events before ingestion you could move this logic from the app into your client before you send the event
+   2. If the app was mainly to send events to a custom URL, we are soon releasing a webhook destination. You can subscribe for updates on the [roadmap](/roadmap)
+   3. If the options above don't work and you were previously paying a substantial amount self-hosting then email us at [hey@posthog.com](mailto:hey@posthog.com) with a link to the public github repo and we can see if it's appropriate as a private cloud app.
```

Comment:
let's not recommend generalising the apps, I don't want to get a lot of review requests for apps, we're not ready for that, this might change in the future, but no distractions for the team currently 🙏 

```suggestion
Have a custom PostHog app on your self-hosted instance that you'd like to move to the cloud? There are several options:
If the app was transforming events before ingestion 
1. (faster option) you could move this logic from the app into your client before you send the event
1. If you can make your app generalizable enough that others can benefit then [submit your app](https://posthog.com/docs/apps/build/tutorial#submitting-your-app) to the store.
   - To make it generalizable you'll want to convert anything which is specific to your configuration into a plugin.json config value

If the app was operating on ingested events
1. (faster option) Convert your app to work as a webhook. We are soon releasing a webhook destination. You can subscribe for updates on the [roadmap](/roadmap)
1. If you can make your app generalizable enough that others can benefit then [submit your app](https://posthog.com/docs/apps/build/tutorial#submitting-your-app) to the store.
   - To make it generalizable you'll want to convert anything which is specific to your configuration into a plugin.json config value

If the options above don't work and you were previously paying a substantial amount self-hosting then email us at [hey@posthog.com](mailto:hey@posthog.com) with a link to the public github repo and we can see if it's appropriate as a private cloud app.
```

Reply:
😮  maybe a topic for the team sync ... I don't think we have the capacity to even support currently existing apps well.

Reply:
We can chat about it during the team sync as well. My opinion on the matter for posterity though:
- We encourage people to make PRs to create generalized Apps (this is good!)
- We don't have any guarantees of reviewing them in x time period. Maybe something the Luigi could do? @pjhul could also take some ownership of this in his free cycles 🤔